auth/oauth.go

```diff
@@ -40,6 +40,21 @@ type OAuthFlow struct {
 	TimeSleep        func(time.Duration)
 }
 
+func detectDeviceFlow(statusCode int, values url.Values) (bool, error) {
+	if statusCode == http.StatusUnauthorized || statusCode == http.StatusForbidden ||
+		statusCode == http.StatusNotFound || statusCode == http.StatusUnprocessableEntity ||
+		(statusCode == http.StatusOK && values == nil) ||
```

Comment:
This seems like a bit of an obfuscated way to handle the case where the `Content-Type` header is incorrect. I don't have strong feelings but it might be more clear to do this check in `ObtainAccessToken`.

Reply:
Agreed; it's not really straightforward. But if you view the expression as "if code was 200, but there were no response values," then it kind of makes sense (I hope?).

We could definitely also do this in `ObtainAccessToken`, but then we're doing Device Flow detection in two different places, which defeats the purpose of extracting it all to one place so it can be unit-tested.